src/python/dxpy/scripts/dx.py

```diff
@@ -2084,26 +2085,56 @@ def upload(args, **kwargs):
         # Called as "dx upload x y --dest /z", z is implicitly a folder, so append a slash to avoid incorrect path
         # resolution.
         args.path += "/"
-
     paths = copy.copy(args.filename)
     for path in paths:
         args.filename = path
         upload_one(args, **kwargs)
 
 upload_seen_paths = set()
+
+def upload_one_v2_file(volume, path, etag, args):
+    import boto3
+    s3 = boto3.client('s3')
+    bucket_name = 'dnanexus-dev-comp14s2' or volume
+    filename = os.path.basename(path)
+
+
+    # Send the file
+    with open(filename, 'rb') as fd:
+        result = s3.put_object(
+            Bucket=bucket_name,
+            Key=path,
+            Body=fd
+        )
+
+    if result['ResponseMetadata']['HTTPStatusCode'] == 200:
+        response = "https://{0}.s3.us-east-2.amCCazonaws.com/{1}".format(bucket, s3_key)
+    else:
+       response = False
+
+    return response
+
 def upload_one(args):
     try_call(process_dataobject_args, args)
 
     args.show_progress = args.show_progress and not args.brief
 
     if args.path is None:
         project = dxpy.WORKSPACE_ID
-        folder = dxpy.config.get('DX_CLI_WD', '/')
+        folder_or_volume = dxpy.config.get('DX_CLI_WD', '/')
         name = None if args.filename == '-' else os.path.basename(args.filename)
     else:
-        project, folder, name, is_v2_path, etag = try_call(resolve_path, args.path)
+        project, folder_or_volume, name, is_v2_path, etag = try_call(resolve_path, args.filename)
+        print(project)
+        print(folder_or_volume)
+        print(name)
+        print(is_v2_path)
+        print(etag)
+        print(args.path)
         if name is None and args.filename != '-':
             name = os.path.basename(args.filename)
+        if is_v2_path:
+            upload_one_v2_file(folder_or_volume, args.path, etag, args)
```

Comment:
We are still gonna call `upload_local_file` after calling `upload_one_v2_file` here, right? 
@kpjensen I am still wondering if it is better to use `upload_local_file` as the common entry point for all uploading (as its name suggests) and add this v2 path uploader inside. Suggestions?

Reply:
we will not be calling upload_local_file after calling upload_one_v2_file I forgot to put a return